x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_overview.tsx

```diff
@@ -55,8 +67,8 @@ export const EnginesOverview: React.FC = () => {
   }, [enginesMeta.page.current]);
 
   useEffect(() => {
-    if (hasPlatinumLicense) loadMetaEngines();
-  }, [hasPlatinumLicense, metaEnginesMeta.page.current]);
+    if (canViewMetaEngines) loadMetaEngines();
+  }, [canViewMetaEngines, metaEnginesMeta.page.current]);
```

Comment:
I would strongly push back against removing `hasPlatinumLicense`. I would leave it alone and drop `canViewMetaEngines`:

```suggestion
    if (hasPlatinumLicense && canViewMetaEngines) loadMetaEngines();
  }, [hasPlatinumLicense, canViewMetaEngines, metaEnginesMeta.page.current]);
```

`hasPlatinumLicense` is an observable that's coming in from Kibana itself and updates dynamically if the user either adds a platinum license or their license expires. That isn't true of `canViewMetaEngines` coming in from the server - so the advantage with `hasPlatinumLicense` is that it's more responsive and we run into slightly less UI jank or permissions issues when a user's platinum license expires.

I'm additionally going to ping @zumwalt here on product permissioning. We removed the `canViewEngines` checks around engine routes because there's no longer a scenario (in a non-SaaS world) where we have users who do not have access to basic engine functionality. I assume this also applies to meta engines; looking at our current role mappings we do not have any roles that distinguish between meta vs source engines, and as such we shouldn't bother to filter for this permission.

Reply:
👍 This is true. The only gate to seeing meta engines should be the presence of a platinum license.

Reply:
I don't like the idea of having multiple ways of checking access to a feature. That's the whole point of these abilities, we have a single source of truth for access to a feature and can update it there.  We're already [checking for the presence of the platinum license](https://github.com/elastic/ent-search/blob/master/loco_moco/lib/loco_moco/ability.rb#L238-L240) to [determine `canViewMetaEngines`/`canManageMetaEngines`](https://github.com/elastic/ent-search/blob/master/loco_moco/lib/loco_moco/ability.rb#L55).  Even if, in this case, we get an advantage of being able to track a variable in a useEffect, I'd still rather completely rely on `AppLogic.values.myRole` for all feature access.

Reply:
nvm I don't really care lmao